monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,159 @@ def get_params(self, image_size):
+            Defaults to ``None``.
+        tile_size: size of the square tile
+            Defaults to ``256``.
+        step: step size
+            Defaults to None (same as tile_size)
+        random_offset: Randomize position of tile grid, instead of starting from the top-left corner
+            Defaults to ``False``.
+        pad_full: pad image to the size evenly divisible by tile_size
+            Defaults to ``False``.
+        background_val: the background constant (e.g. 255 for white background)
+            Defaults to ``255``.
+        filter_mode: mode must be in ["min", "max", None]. If total number of tiles is more then tile_size,
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for None) subset
+            Defaults to ``min`` (which assumes background is white, high value)
+
+    """
+
+    def __init__(
+        self,
+        tile_count: Optional[int] = None,
+        tile_size: int = 256,
+        step: Optional[int] = None,
+        random_offset: bool = False,
+        pad_full: bool = False,
+        background_val: int = 255,
+        filter_mode: Optional[str] = "min",
+    ):
+        self.tile_count = tile_count
+        self.tile_size = tile_size
+        self.step = step
+        self.random_offset = random_offset
+        self.pad_full = pad_full
+        self.background_val = background_val
+        self.filter_mode = filter_mode
+
+        # self.tile_all = (self.tile_count is None)
+
+        # if self.tile_count is None:
+        #     self.tile_count = max((44 * 256 ** 2) // (tile_size ** 2), 1)
+
+        if self.step is None:
+            self.step = self.tile_size  # non-overlapping grid
+
+        self.offset = (0, 0)
+        self.random_idxs = [0]
+
+    def randomize(self, img_size: Sequence[int]) -> None:
+
+        c, h, w = img_size
+        # tile_count: int = self.tile_count  # type: ignore
+        tile_step: int = self.step  # type: ignore
+
+        if self.random_offset:
+            pad_h = h % self.tile_size
+            pad_w = w % self.tile_size
+            if pad_h > 0 and pad_w > 0:
+                self.offset = (self.R.randint(pad_h), self.R.randint(pad_w))
+                h = h - self.offset[0]
+                w = w - self.offset[1]
+            else:
+                self.offset = (0, 0)
+
+        if self.pad_full:
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            h = h + pad_h
+            w = w + pad_w
+
+        h_n = (h - self.tile_size + tile_step) // tile_step
+        w_n = (w - self.tile_size + tile_step) // tile_step
+        tile_total = h_n * w_n
+
+        if self.tile_count is not None and tile_total > self.tile_count:
+            self.random_idxs = self.R.choice(range(tile_total), self.tile_count, replace=False)  # type: ignore
```

Comment:
What is the type error that you are getting here?
I can guess that `choice` is returning `ndarray` and you have initialized `self.random_idxs` with list. So they are not of the same type. If that's a problem, instead of ignoring the error, try to use `tolist` method to convert ndarray to list.

Reply:
ok